File: silver/1316/c++/main.cpp
#include <iostream>
#include <string>

int main()
{
    int num, count = 0;
    char choiceText;
    std::string text;
    std::cin >> num;

    int alphabet[26] = {0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0};


    for (int i = 0; i < num; i++) {
        std::cin >> text;
        choiceText = text[0];
        int j;
        for (j = 0; j < text.size(); j++) {
            if (alphabet[(static_cast<int>(text[j]) - 97)] == 1) {
                j = 0;
                break;
            }
            if (choiceText != text[j]) {
                alphabet[(static_cast<int>(choiceText) - 97)] = 1;
                choiceText = text[j];
            }
        }

        if (j > 0) {
            count++;
        }
        for (int k = 0; k < 26; k++) {
            alphabet[k] = 0;
        }
    }
    std::cout << count;

    return 0;
}

File: silver/1316/python/main.py

if (__name__ == "__main__"):
    num = int(input())
    count = 0
    alphabet = [0 for _ in range(26)]
    for i in range(num):
        text = input()
        choice = text[0]

        isPass = True
        for j in range(len(text)):
            if (alphabet[ord(text[j]) - 97] == 1):
                isPass = False
                break
            if (choice != text[j]):
                alphabet[ord(choice) - 97] = 1
                choice = text[j]

        if (isPass):
            count += 1
        alphabet = [0 for _ in range(26)]
    print(count)
